workflow/workflow.go

```diff
@@ -325,31 +349,91 @@ func (l *loopState) onStageComplete(stepID string, previousStage *string, previo
 	l.notifySteps()
 }
 
+// Marks the outputs of that stage unresolvable.
+// Optionally skip an output that should not unresolve.
+// Does not (un)resolve the non-output node for that stage.
+// To prevent a deadlock, this `notifySteps()` should be called at some point after this is called.
+func (l *loopState) unresolveOutputs(stepID string, stageID string, skippedOutput *string) {
```

Comment:
so the function name `unresolveOutputs` means to mark outputs as unresolvable?

Reply:
I changed the name to be less ambiguous.